### PR TITLE
[Currency] Documentation update of Currency component

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q             | A
| ------------- | ---
| Doc fix?      | yes
| New docs?     | no
| Fixed tickets | -

### CODE REVIEW

File: components/Currency/examples/contexts.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+The **CurrencyContext** gives you the ability to quickly
+set and retrieve currency names from a given **Storage**.
+
+.. note::
+   You can use any storage you would like as long as it implements the ``StorageInterface``
+
+In this example let's use the default **SessionStorage**.
+
+.. code-block:: php
+
```
Comment (TheMadeleine) [inline components/Currency/examples/contexts.rst]: I'm missing the ``<?php`` line there :D

Comment (pamil) [inline components/Currency/examples/contexts.rst]: IMHO we don't need `<?php` at all in these listings :)

File: components/Currency/basic_usage.rst [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Basic Usage
+===========
+
+.. toctree::
+   :maxdepth: 1
+
+   examples/contexts
+   examples/converters
+   examples/importers
+   examples/providers
```
Comment (tuka217) [inline components/Currency/basic_usage.rst]: All basic usage should be in one file.

File: components/Currency/interfaces.rst [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+==========
+
+.. toctree::
+   :maxdepth: 2
+
+   interfaces/model_interfaces
+   interfaces/context_interfaces
+   interfaces/converter_interfaces
+   interfaces/importer_interfaces
+   interfaces/provider_interfaces
```
Comment (tuka217) [inline components/Currency/interfaces.rst]: All interfaces should be in one file.

File: components/Currency/basic_usage.rst [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+.. _Open Exchange Rates: http://openexchangerates.org
+
+.. code-block:: php
+
+   use Sylius\Component\Currency\Importer\OpenExchangeRatesImporter;
+   // also use your repository of currencies
+   // and an object manager
+
+   $manager = // your manager
+   $repository = // your repository
```
Comment (tuka217) [inline components/Currency/basic_usage.rst]: Maybe create in memory repository.

File: components/Currency/basic_usage.rst [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+The **CurrencyProvider** allows you to get all available currencies.
+
+.. code-block:: php
+
+   <?php
+
+   use Sylius\Component\Currency\Provider\CurrencyProvider;
+   // also use your repository of currencies
+
+   $currencyRepository = // your repository
```
Comment (tuka217) [inline components/Currency/basic_usage.rst]: Same as above.

File: components/Currency/basic_usage.rst [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+.. _Intl: http://symfony.com/doc/current/components/intl.html
+
+.. code-block:: php
+
+   <?php
+
+   use Sylius\Component\Currency\Model\Currency;
+
+   $currency = new Currency();
+
```
Comment (pjedrzejewski) [inline components/Currency/basic_usage.rst]: No need for the blank line, I think.

File: components/Currency/basic_usage.rst [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+.. note::
+   The output of ``getName`` may vary as the name is generated accordingly to the set locale.
+
+.. _component_currency_context_currency-context:
+
+CurrencyContext
+---------------
+
+The **CurrencyContext** gives you the ability to quickly
+set and retrieve currency names from a given :doc:`/components/Storage/index`.
```
Comment (pjedrzejewski) [inline components/Currency/basic_usage.rst]: Actually, CurrenctContext is responsible for keeping the currently selected/used currency. :)

File: components/Currency/basic_usage.rst [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+Importers
+---------
+
+Importers allow you to get up-to-date exchange rates from given provider.
+The rates are set from your chosen base currency to all other managed currencies.
+
+.. _component_currency_importer_european-central-bank-importer:
+
+EuropeanCentralBankImporter
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (pjedrzejewski) [inline components/Currency/basic_usage.rst]: I think we should move this and AbstractImporter into separate section -> importers or Importing exchange rates.

File: components/Currency/basic_usage.rst [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+
+.. _European Central Bank: http://www.ecb.int
+
+.. code-block:: php
+
+   <?php
+
+   use Sylius\Component\Currency\Importer\EuropeanCentralBankImporter;
+   use Sylius\Component\Resource\Repository\InMemoryRepository;
+   // also use an object manager
```
Comment (pjedrzejewski) [inline components/Currency/basic_usage.rst]: Let's specify here ObjectRepository interface from Doctrine Persistence. Hm?

**Comment by pjedrzejewski** [comment]
Really cool piece of docs, nice job. :+1:

**Comment by pjedrzejewski** [comment]
Thanks Jasiek!

**Pull request merged**